src/deepEqual.js

```diff
@@ -21,6 +21,35 @@
  */
 function deepEqual(a, b) {
   // write code here
+  let typeOfA = typeof a;
+  let typeOfB = typeof b;
+
+  // If data equal...
+  if (typeOfA === typeOfB && a === b) {
+    return true;
+  // Only objects can be toataly non equal
+  } else if (typeOfA !== 'object' || typeOfB !== 'object') {
+    return false;
+  }
+
+  if (a !== null && b !== null) {
+    // Objects are non equal if there are different amount of keys in them
+    if (Object.keys(a).length !== Object.keys(b).length) { // Если количество ключей объектов не раное - они точно не равны.
+      return false;
+    }
+    for (let key in a) {
+      if (!(key in b)) {
+        return false;
+      }
+      if (!deepEqual(a[key], b[key])) {
+        return false;
+      }
+    }
+  } else {
+    return false;
+  }
+
+  return true; // Если программа выполнилась до этого момента - сущности равны.
```

Comment:
remove comment in russian. It's actually not useful, I would say don't write it at all.